prov/rxm/src/rxm_cq.c

```diff
@@ -460,16 +461,17 @@ ssize_t rxm_cq_handle_rndv(struct rxm_rx_buf *rx_buf)
 	struct rxm_rx_buf *new_rx_buf;
 	int ret = 0;
 
-	rx_buf->repost = 0;
-
 	/* En-queue new rx buf to be posted ASAP so that we don't block any
-	 * incoming messages. RNDV processing can take a while. */
-	new_rx_buf = rxm_rx_buf_alloc(rx_buf->ep, rx_buf->msg_ep, 1);
-	if (!new_rx_buf)
-		return -FI_ENOMEM;
+	* incoming messages. RNDV processing can take a while. */
+	if (rx_buf->repost) {
+		rx_buf->repost = 0;
+		new_rx_buf = rxm_rx_buf_alloc(rx_buf->ep, rx_buf->msg_ep, 1);
+		if (!new_rx_buf)
+			return -FI_ENOMEM;
 
-	dlist_insert_tail(&new_rx_buf->repost_entry,
-			  &new_rx_buf->ep->repost_ready_list);
+		dlist_insert_tail(&new_rx_buf->repost_entry,
+				  &new_rx_buf->ep->repost_ready_list);
+	}
```

Comment:
It looks like we can update all 3 places with this same logic captured above.  Just need to find a good name for the function.